src/util/request_proxy.rs

```diff
@@ -1,26 +1,53 @@
 use std::{io::Read, net::SocketAddr};
 
-use conduit::Request;
+use conduit::{Method, Request};
 use conduit_hyper::semver;
 
+type RequestMutRef<'a> = &'a mut (dyn Request + 'a);
+
 // Can't derive Debug because of Request.
 #[allow(missing_debug_implementations)]
 pub struct RequestProxy<'a> {
-    pub other: &'a mut (dyn Request + 'a),
-    pub path: Option<&'a str>,
-    pub method: Option<conduit::Method>,
+    other: RequestMutRef<'a>,
+    path: Option<&'a str>,
+    method: Option<conduit::Method>,
+}
+
+impl<'a> RequestProxy<'a> {
+    pub(crate) fn rewrite_path(req: RequestMutRef<'a>, path: &'a str) -> Self {
```

Comment:
You could consider adding a one-sentence doc comment here, and for the other constructor as well.